ledger/src/blockstore.rs

```diff
@@ -1486,13 +1589,30 @@ impl Blockstore {
             write_batch,
             shred_source,
         )?;
-        merkle_root_metas
-            .entry(erasure_set)
-            .or_insert(WorkingEntry::Dirty(MerkleRootMeta::from_shred(&shred)));
+
+        let mut new_merkle_root_meta = false;
+        merkle_root_metas.entry(erasure_set).or_insert_with(|| {
+            new_merkle_root_meta = true;
+            WorkingEntry::Dirty(MerkleRootMeta::from_shred(&shred))
+        });
```

Comment:
I personally don't like this `let mut ...` + mutation pattern. It is much easier to make mistakes and introduce bugs with mutable variables. Instead
```rust
let new_merkle_root_meta = match merkle_root_metas.entry(erasure_set) {
   Entry::Vacant(entry) => {
      entry.insert(...);
      true
   }, 
   Entry::Occupied(_) => false,
}
```